src/lib.rs

```diff
@@ -234,16 +247,28 @@ pub struct GenerationConfig<'a> {
     pub once_common_structs: bool,
     /// Generate the "ConnectionType" type only once in a "common.rs" file
     pub once_connection_type: bool,
+    /// Prefixes to treat tables as readonly
+    pub readonly_prefixes: Vec<String>,
+    /// Suffixes to treat tables as readonly
+    pub readonly_suffixes: Vec<String>,
 }
 
 impl GenerationConfig<'_> {
     pub fn table(&self, name: &str) -> TableOptions<'_> {
-        let t = self
+        let table = self
             .table_options
             .get(name)
             .unwrap_or(&self.default_table_options);
 
-        t.apply_defaults(&self.default_table_options)
+        let mut table = table.apply_defaults(&self.default_table_options);
+
+        if self.readonly_prefixes.iter().any(|v| name.starts_with(v))
+            || self.readonly_suffixes.iter().any(|v| name.ends_with(v))
+        {
+            table.set_read_only(true);
+        }
+
+        table
```

Comment:
I think we should have a computation step in the beginning of `lib.rs#generate_files()` that does some of these pre-generation computations, including aplying the default table options.

Going to create a new issue for this